app/src/main/java/com/example/cocktaildb/screen/createrecipe/CreateRecipePresenter.kt

```diff
@@ -167,6 +171,62 @@ class CreateRecipePresenter(
         }
     }
 
+    private suspend fun saveRecipeImage(recipeId: String, imageUrl: String) {
+        try {
+            Log.d("CreateRecipePresenter", "Starting to save image with URL: $imageUrl")
+
+            val uploadResult = if (imageUrl.startsWith("content://")) {
+                Log.d("CreateRecipePresenter", "Uploading content:// URI to ImgBB")
+                val sourceUri = Uri.parse(imageUrl)
+                firebaseRepository.uploadRecipeImage(context, sourceUri, recipeId)
+            } else if (firebaseRepository.isImgBBUrl(imageUrl)) {
+                Log.d("CreateRecipePresenter", "Using existing ImgBB URL: $imageUrl")
+                Result.success(imageUrl)
+            } else {
+                Log.w("CreateRecipePresenter", "Unknown image URL format: $imageUrl, attempting upload")
+                val sourceUri = Uri.parse(imageUrl)
+                firebaseRepository.uploadRecipeImage(context, sourceUri, recipeId)
+            }
+
+            if (uploadResult.isSuccess) {
+                val downloadUrl = uploadResult.getOrThrow()
+                Log.d("CreateRecipePresenter", "Image uploaded successfully to ImgBB: $downloadUrl")
+                
+                val recipeImage = RecipeImage(
+                    id = "",
+                    recipeId = recipeId,
+                    imageUrl = downloadUrl,
+                    isPrimary = true,
+                    caption = "",
+                    uploadedAt = System.currentTimeMillis()
+                )
+                
+                Log.d("CreateRecipePresenter", "Saving image record to Firestore")
+                val result = firebaseRepository.addRecipeImage(recipeImage)
+                if (result.isSuccess) {
+                    Log.d("CreateRecipePresenter", "Recipe image saved successfully")
+                } else {
+                    val error = result.exceptionOrNull()
+                    Log.e("CreateRecipePresenter", "Failed to save image record: ${error?.message}")
+                    withContext(Dispatchers.Main) {
+                        view?.showError("Warning: Image uploaded but failed to save record")

```

Comment:
```suggestion
                        view?.showError("Warning: Image uploaded but failed to save record")
```
hard code, dùng string resource